src/libraries/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.TryReadStatusFile.cs

```diff
@@ -149,21 +150,69 @@ internal static bool TryParseStatusFile(string statusFilePath, out ParsedStatus
             return true;
         }
 
-        private static bool TryReadFile(string filePath, ReusableTextReader reusableReader, [NotNullWhen(true)] out string? fileContents)
+        private static bool TryReadFile(string path, [NotNullWhen(true)] out string? contents)
         {
+            Debug.Assert(!string.IsNullOrEmpty(path));
+
+            Encoding encoding = Encoding.UTF8;
+            Decoder decoder = encoding.GetDecoder();
+
+            char[]? chars = null;
+            byte[]? bytes = null;
+            const int size = 4096;
             try
             {
-                using (var fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false))
+                using var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read, bufferSize: 1, useAsync: false);
+
+                bytes = ArrayPool<byte>.Shared.Rent(size);
+                chars = ArrayPool<char>.Shared.Rent(encoding.GetMaxCharCount(size));
+                int charCount = 0;
+
+                while (true)
                 {
-                    fileContents = reusableReader.ReadAllText(fileStream);
-                    return true;
+                    int count = size;
+                    int index = 0;
+                    while (count > 0)
+                    {
+                        int read = fileStream.Read(bytes, 0, count);
+                        if (read == 0)
+                        {
+                            contents = new string(chars, 0, charCount);
+                            return true;
+                        }
+
+                        if (charCount + read > chars.Length)
+                        {
+                            char[] temp = ArrayPool<char>.Shared.Rent(encoding.GetMaxCharCount(chars.Length * 2));
```

Comment:
It should be better to keep growing the byte array and do the encoding at the end using `UTF8Encoding.GetString(Byte[], Int32, Int32)`.

Reply:
Removed `char[]` and internal loop in favor of GetString.